src/nvim/macros.h

```diff
@@ -35,8 +35,8 @@
 #define LINEEMPTY(p) (*ml_get(p) == NUL)
 
 /// BUFEMPTY() - return TRUE if the current buffer is empty
-#define BUFEMPTY() (curbuf->b_ml.ml_line_count == 1 && *ml_get((linenr_T)1) == \
-                    NUL)
+#define BUFEMPTY(buf) ((buf)->b_ml.ml_line_count == 1 \
```

Comment:
if you feel like refactor I think this should be `static inline bool buf_empty(buf_T *buf)` in `buffer.h` (but if not we can do that later).

Reply:
Sounds good to me then

Reply:
Done